src/wordlist.c

```diff
@@ -550,6 +552,24 @@ void do_wordlist_crack(struct db_main *db, const char *name, int rules)
 	}
 
 	if (name) {
+		struct stat st;
+
+		if (!(word_file = jtr_fopen(path_expand(name), "rb")))
+			pexit(STR_MACRO(jtr_fopen)": %s", path_expand(name));
+
+		if (fstat(fileno(word_file), &st))
+			pexit("fstat");
+
+		if ((st.st_mode & S_IFMT) == S_IFIFO)
+			file_is_fifo = 1;
```

Comment:
I think we should reset `file_is_fifo` to 0 on new calls to `do_wordlist_crack`. Right now it's just one such call per `john` invocation, but that can change.